odxtools/cli/_print_utils.py

```diff
@@ -53,31 +63,197 @@ def print_diagnostic_service(
         print(f"  Enabled-Audiences: {', '.join(enabled_audiences_short_names)}")
 
     if print_params:
-        assert service.request is not None
-        assert service.positive_responses is not None
-        assert service.negative_responses is not None
+        print_service_parameters(
+            service,
+            allow_unknown_bit_lengths=allow_unknown_bit_lengths,
+            print_objects=print_objects)
 
-        print(f"  Message format of a request:")
-        service.request.print_message_format(
-            indent=3, allow_unknown_lengths=allow_unknown_bit_lengths)
 
-        print(f"  Number of positive responses: {len(service.positive_responses)}")
-        if len(service.positive_responses) == 1:
-            resp = service.positive_responses[0]
+def print_service_parameters(service: DiagService,
+                             allow_unknown_bit_lengths: bool = False,
+                             print_objects: bool = False) -> None:
+    # prints parameter details of request, posivite response and negative response of diagnostic service
 
-            print(f"  Message format of a positive response:")
-            resp.print_message_format(indent=3, allow_unknown_lengths=allow_unknown_bit_lengths)
+    assert service.request is not None
+    assert service.positive_responses is not None
+    assert service.negative_responses is not None
 
-        print(f"  Number of negative responses: {len(service.negative_responses)}")
-        if len(service.negative_responses) == 1:
-            resp = service.negative_responses[0]
+    # Request
+    print(f"\n  [yellow]Request Properties[/yellow]:")
+    print(f"   Request Name: {service.request.short_name}")
+    if not getattr(service.request, "required_parameters", None):
+        if len(service()) > 24:
+            print(
+                f"   Byte-Array: {service()[:24]!r}...\tHex-String: 0x{service().hex().upper()[:24]}..."
```

Comment:
tip: use ellipse character instead of three dots

https://www.compart.com/en/unicode/U+2026

Reply:
I guess three dots are better; it keeps the output ASCII as long as no non-ascii characters are featured by in the dataset...